Testing/UnitTests/xASL_ut_function_xASL_str2num.m

```diff
@@ -0,0 +1,95 @@
+function UnitTest = xASL_ut_function_xASL_str2num(TestRepository)
+%xASL_ut_function_xASL_str2num Individual unit test for xASL_str2num
+%
+% INPUT:        TestRepository - Path to test repository.
+%
+% OUTPUT:       UnitTest  - Test structure
+%               name      - Name of tested module or submodule (char array)
+%               unit      - Insert one of the following: 'Module', 'Submodule' or 'Function'
+%               passed    - Result of all subtests combined (true or false)
+%               test      - Structure with individual subtest results
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  Should be run using xASL_ut_UnitTesting.
+%
+% EXAMPLE:      UnitTests(1) = xASL_ut_function_xASL_str2num(TestRepository);
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% Copyright 2015-2021 ExploreASL
+
+
+%% Test run 1
+
+% Give your individual subtest a name
+UnitTest.tests(1).testname = 'Number types: integer, double, positive and negative';
+
+% Start the test
+testTime = tic;
+
+% Run your test here
+numOutA = xASL_str2num('123');
```

Comment:
also stuff like 10e6 , 5.3e-6, .12, and some nonsense stuff - to see if it doesn't crash if a non-number is provided.

Reply:
01dfe87